agoora-kafka-agent/src/main/java/io/spoud/agoora/agents/kafka/schema/SchemaService.java

```diff
@@ -28,47 +29,82 @@ public Map<String, String> update(String topicName, String dataPortId) {
     Map<String, String> properties = new HashMap<>();
     try {
       for (SchemaRegistryClient schemaRegistry : schemaRegistries) {
-        schemaRegistry
+
+        Schema valueSchema = schemaRegistry
             .getLatestSchemaForTopic(topicName, KafkaStreamPart.VALUE)
-            .ifPresentOrElse(
-                schema -> {
+            .orElseGet(
+                () -> {
                   LOG.debug(
-                      "One schema found for the topic {} in the registry {}",
-                      topicName,
+                      "No schema found for topic {} in the registry {}", topicName,
                       schemaRegistry.getDeepDiveToolUrl(topicName));
+                  return null;
+                });
 
-                  try {
-                    Schema saved =
-                        schemaClient.saveSchema(
-                            ResourceEntity.Type.DATA_PORT,
-                            dataPortId,
-                            kafkaAgentConfig
-                                .getTransport()
-                                .getAgooraPathObject()
-                                .getResourceGroupPath(),
-                            schema.getContent(),
-                            SchemaSource.Type.REGISTRY,
-                            schema.getEncoding());
-                    LOG.info("Successfully saved schema for topic {} with uuid '{}'", topicName, saved.getId());
-                  } catch (StatusRuntimeException ex) {
-                    LOG.error("Unable to save schema '{}' for topic {}", schema, topicName, ex);
-                  }
-
-                  schemaRegistry
-                      .getDeepDiveToolUrl(topicName)
-                      .ifPresent(
-                          url ->
-                              properties.put(
-                                  Constants.PROPETIES_DEEP_DIVE_TOOL_SCHEMA_REGISTRY, url));
-                },
+        Schema keySchema = schemaRegistry
+            .getLatestSchemaForTopic(topicName, KafkaStreamPart.KEY)
+            .orElseGet(
                 () -> {
                   LOG.debug(
-                      "No schema found for topic {} in the registry {}", topicName, schemaRegistry.getDeepDiveToolUrl(topicName));
+                      "No schema found for topic {} in the registry {}", topicName,
+                      schemaRegistry.getDeepDiveToolUrl(topicName));
+                  return null;
                 });
+
+        saveSchema(topicName, dataPortId, properties, schemaRegistry, valueSchema, keySchema);
+
+
       }
     } catch (Exception ex) {
       LOG.error("Unable to update schema", ex);
     }
     return properties;
   }
+
+  private void saveSchema(String topicName, String dataPortId, Map<String, String> properties, SchemaRegistryClient schemaRegistry, Schema valueSchema, Schema keySchema) {
+
+    // TODO: accept schema with only KEY schema/encoding
```

Comment:
Why is it not resolved?

Reply:
sdm-schema table does not accept null content/type for value schema